tbtc-ts/src/bitcoin.ts

```diff
@@ -110,13 +185,128 @@ export interface Client {
+
+  /**
+   * Gets height of the latest mined block.
+   * @return Height of the last mined block.
+   */
+  latestBlockHeight(): Promise<number>
+
+  /**
+   * Gets concatenated chunk of block headers built on a starting block.
+   * @param blockHeight - Starting block height.
+   * @param confirmations -  Number of confirmations (subsequent blocks) built
+   *                         on the starting block.
+   * @return Concatenation of block headers in a hexadecimal format.
+   */
+  getHeadersChain(blockHeight: number, confirmations: number): Promise<string>
+
+  /**
+   * Get proof of transaction inclusion in the block.
+   * @param txHash - Hash of a transaction.
+   * @param blockHeight - Height of the block where transaction was confirmed.
+   * @return Transaction inclusion proof in hexadecimal form.
+   */
+  getTransactionMerkle(
+    txHash: string,
+    blockHeight: number
+  ): Promise<TransactionMerkleBranch>
+
   /**
    * Broadcasts the given transaction over the network.
    * @param transaction - Transaction to broadcast.
    */
   broadcast(transaction: RawTransaction): Promise<void>
 }
 
+/**
+ * Converts buffered data to hex string
+ * @param bufferWriter - Buffer writer containing data to be converted
+ * @returns Buffer converted to string
+ */
+function toHex(bufferWriter: StaticWriter | BufferWriter): string {
+  return bufferWriter.render().toString("hex")
+}
+
+/**
+ * Gets transaction version from a transaction and returns it in as hex string.
+ * @param tx - Transaction to extract the version from.
+ * @returns Transaction version as hex string.
+ */
+function getTxVersion(tx: TX): string {
+  const buffer = bufio.write()
+  buffer.writeU32(tx.version)
+  return toHex(buffer)
+}
+
+/**
+ * Gets locktime from a transaction and returns it in as hex string.
+ * @param tx - Transaction to extract the locktime from.
+ * @returns Transaction locktime as hex string.
+ */
+function getTxLocktime(tx: TX): string {
+  const buffer = bufio.write()
+  buffer.writeU32(tx.locktime)
+  return toHex(buffer)
+}
+
+/**
+ * Converts a vector of elements into a single hex string
+ * @param elements - Elements to be converted to hex string
+ * @returns Hex string representation of vector elements
+ */
+function vectorToRaw(elements: any[]): string {
+  const buffer = bufio.write()
+  buffer.writeVarint(elements.length)
+
+  for (const element of elements) {
+    element.toWriter(buffer)
+  }
+
+  return toHex(buffer)
+}
+
+/**
+ * Gets vector of inputs as hex string from a transaction
+ * @param tx - Transaction
+ * @returns Hex string representation of transaction input vector
+ */
+function getTxInputVector(tx: TX): string {
+  return vectorToRaw(tx.inputs)
+}
+
+/**
+ * Gets vector of outputs as hex string from a transaction
+ * @param tx - Transaction
+ * @returns Hex string representation of transaction output vector
+ */
+function getTxOutputVector(tx: TX): string {
+  return vectorToRaw(tx.outputs)
+}
```

Comment:
I understand the purpose but those helpers are strictly related to transaction decomposition and can live under `decomposeRawTransaction` function to make it clear they are not part of the `bitcoin.ts` module interface. My proposition:
```
export function decomposeRawTransaction(rawTransaction: string): Info {
  const getTxVersion = (tx: TX) => {
     const buffer = bufio.write()
     buffer.writeU32(tx.version)
     return toHex(buffer)
  }

  (...other functions...)

  const tx = TX.fromRaw(Buffer.from(rawTransaction, "hex"), null)

  return {
    txVersion: getTxVersion(tx),
    txInputVector: getTxInputVector(tx),
    txOutputVector: getTxOutputVector(tx),
    txLocktime: getTxLocktime(tx),
  }
}
```

Reply:
Good idea. Done.